app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -37,6 +38,24 @@
     private EditText mPortEdit;
     private TextView mResultText;
 
+    public final int getPort() {
+        String port = mPortEdit.getText().toString();
+        if (TextUtils.isEmpty(port)) {
+            Toast.makeText(this, "You did not enter a port", Toast.LENGTH_SHORT).show();
```

Comment:
You're right. I need to throw a throwable object. I chose MissingFormatArgumentException. It looks appropriate for this.